src/components/content/policies/PolicyDetails.tsx

```diff
@@ -0,0 +1,95 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ */
+
+import '../../../styles/app.css';
+import React, { useEffect, useState } from 'react';
+import usePoliciesManagementServiceDetailQuery from './details/usePoliciesManagementServiceDetailQuery';
+import { AbstractCredentialInfo, CloudServiceProvider, PolicyVo } from '../../../xpanse-api/generated';
+import { ColumnsType } from 'antd/es/table';
+import { Button, Image, Popover, Space, Table } from 'antd';
+import PolicyServiceDetailsError from './details/PolicyServiceDetailsError';
+import { PolicyServiceDetailsSkeleton } from './details/PolicyServiceDetailsSkeleton';
+import { cspMap } from '../common/csp/CspLogo';
+
+export const PolicyDetails = ({ id }: { id: string }): React.JSX.Element => {
+    const policyDetailsQuery = usePoliciesManagementServiceDetailQuery(id);
+    const [policyService, setPolicyService] = useState<PolicyVo[]>([]);
+
+    useEffect(() => {
+        if (policyDetailsQuery.isSuccess) {
+            setPolicyService([policyDetailsQuery.data]);
+        }
+    }, [policyDetailsQuery.isSuccess, policyDetailsQuery.data]);
+
+    if (policyDetailsQuery.isError) {
+        return <PolicyServiceDetailsError error={policyDetailsQuery.error} />;
+    }
+
+    if (policyDetailsQuery.isPending || policyDetailsQuery.isFetching) {
+        return <PolicyServiceDetailsSkeleton />;
+    }
+
+    const columns: ColumnsType<PolicyVo> = [
+        {
+            title: 'Policy ID',
+            dataIndex: 'id',
+        },
+        {
+            title: 'CSP',
+            dataIndex: 'csp',
+            render: (csp: AbstractCredentialInfo.csp, _) => {
```

Comment:
AbstractCredentialInfo should not be used here. Please check for CSP enum in policy related data models.